### PR TITLE
add support for Quantize round mode

### PR DESCRIPTION
**Pull request by adstraw**

### CODE REVIEW

File: src/ngraph/runtime/cpu/cpu_emitter.cpp [hunk 1]
```diff
@@ -4621,15 +4621,17 @@ namespace ngraph
...
                 writer << "            " << args[0].get_name() << ",\n";
                 writer << "            " << args[1].get_name() << ",\n";
                 writer << "            " << args[2].get_name() << ",\n";
                 writer << "            " << out[0].get_name() << ",\n";
                 writer << "            {" << join(args[0].get_shape()) << "},\n";
                 writer << "            {" << join(args[1].get_shape()) << "},\n";
-                writer << "            {" << join(quantize->get_axes()) << "});\n";
+                writer << "            {" << join(quantize->get_axes()) << "},\n";
+                writer << "            static_cast<op::Quantize::RoundMode>("
+                       << static_cast<int>(quantize->get_round_mode()) << "));\n";
```
Comment (adstraw) [inline src/ngraph/runtime/cpu/cpu_emitter.cpp]: I'm not positive this is the correct way to do this and also not sure we are hitting this case with unit tests.  Can you comment on this @nbpatel?

File: src/ngraph/runtime/reference/quantize.hpp [hunk 2]
```diff
@@ -44,11 +47,64 @@ namespace ngraph
...
-                    // apply scale and offset
-                    REAL qvalue =
-                        std::round(input[input_transform.index(input_coord)] /
-                                   scale[scale_offset_transform.index(scale_offset_coord)]) +
-                        offset[scale_offset_transform.index(scale_offset_coord)];
+                    // apply scale
+                    REAL qvalue = input[input_transform.index(input_coord)] /
+                                  scale[scale_offset_transform.index(scale_offset_coord)];
+
+                    auto fe_round_mode = std::fegetround();
```
Comment (diyessi) [inline src/ngraph/runtime/reference/quantize.hpp]: I'd feel more comfortable if you guarded this with RAII. If this setting manages to escape the function, it's really really hard to figure out why the math has become subtly different,

File: src/ngraph/runtime/reference/quantize.hpp [hunk 3]
```diff
@@ -44,11 +47,64 @@ namespace ngraph
...
+
+                    // round
+                    if (round_mode == op::Quantize::RoundMode::HALF_AWAY_FROM_ZERO)
```
Comment (rkimballn1) [inline src/ngraph/runtime/reference/quantize.hpp]: You should make this a switch and add gcc flags inline to make sure you cover all cases.

File: doc/sphinx/source/ops/quantize.rst [hunk 4]
```diff
@@ -40,11 +40,49 @@ Attributes
 +-------------------------------+----------------------------------------------------------------+
 | ``axes``                      | Axis positions on which ``scale`` and ``offset`` are specified |
 +-------------------------------+----------------------------------------------------------------+
-| ``round_mode``                | Refer to ``/src/ngraph/op/quantize.hpp``                       |
+| ``round_mode``                | *ROUND_NEAREST_TOWARD_INFINITY:*                               |
+|                               | round to nearest integer                                       |
+|                               | in case of two equally near integers round away from zero e.g. |
```
Comment (indie) [inline doc/sphinx/source/ops/quantize.rst]: in case of two integers being equidistant,  round away...

File: doc/sphinx/source/ops/quantize.rst [hunk 5]
```diff
@@ -40,11 +40,49 @@ Attributes
...
+|                               | 2.5 -> 3                                                       |
+|                               | -3.5 -> -4                                                     |
+|                               |                                                                |
+|                               | *ROUND_NEAREST_TOWARD_ZERO:*                                   |
+|                               | round to nearest integer                                       |
+|                               | in case of two equally near integers round toward zero e.g.    |
```
Comment (indie) [inline doc/sphinx/source/ops/quantize.rst]: "in case of two integers being equidistant, round toward zero"

**Pull request merged**